core/src/main/java/com/datastax/oss/driver/internal/core/metadata/schema/queries/CassandraSchemaQueries.java

```diff
@@ -73,6 +81,10 @@ protected CassandraSchemaQueries(
             DefaultDriverOption.METADATA_SCHEMA_REFRESHED_KEYSPACES, Collections.emptyList());
     assert refreshedKeyspaces != null; // per the default value
     this.keyspaceFilter = KeyspaceFilter.newInstance(logPrefix, refreshedKeyspaces);
+    this.usingTimeoutClause =
```

Comment:
Q: shouldn't USING TIMEOUT be only applied if METADATA_SCHEMA_REQUEST_TIMEOUT has some value? And if the default will have 0, then we can keep both backward compat. and Cassandra compat. and yet satisfy the ask of this issue, no?

Reply:
This value was originally used as a driver side timeout only and by default it's set to 2 seconds. If set to 0 I believe every query would instantly time out driver-side. 
If i went back to introducing a new config value for this and made it the way you say, then yes it would satisfy the ask of this issue, but I still would have to run a test query for Scylla and adjust tests accordingly. Unless we shift the responsibility to user side and say "use this only if you know your cluster supports this". Tests would require only adding mocks for the config in that case.

Reply:
I think (and others should chime in here) that it should be a Scylla only config, that should not be set by default (and if it's 0, we simply do not use it - or perhaps - though I dislike it, you can use -1 as disable and 0 as infinite - a bit ugly if you ask me). We will ask customers to use this value IF they have a large schema and schema pulls fail and retried (a log about this could be useful as well!). Then we are good from all sides - Cassandra wouldn't use this by default and there is no change in the default behavior.

Reply:
I think it would be weird to have different timeout driver and server side (is there a use case for that?), but there's a middle ground here with adding boolean flag for enabling Using Timeout clause. If enabled driver will pass the same timeout to query strings.

Reply:
The whole point of the feature here is that *schema* pulls will have their own, specific (longer...) timeout values.

Reply:
We do have that distinction already on the driver side. I thought the problem was that timeout is not being overridden on Scylla side by adding `USING TIMEOUT` to queries which causes large schema pulls to time out on Scylla side.

For global limits we have `REQUEST_TIMEOUT` and for schema we have `METADATA_SCHEMA_REQUEST_TIMEOUT` which is already implemented.
I may have caused some confusion by originally introducing a new very similarly named config for schema timeouts intended for server-side but I've removed it already.

Reply:
So I'm somewhat confused what we are doing in this PR. Perhaps a quick call is needed.

Reply:
(in short, why not add USING TIMEOUT to METADATA_SCHEMA_REQUEST_TIMEOUT ? This I think it is the original ask)

Reply:
I think this is what we're doing now: adding `USING TIMEOUT` string to schema query strings with the value of `METADATA_SCHEMA_REQUEST_TIMEOUT` seconds.
The problem with implementation was that `USING TIMEOUT` is an extension to CQL not present in every Scylla version and not present in Cassandra at all. Adding it when not supported will cause queries to fail.
Originally I've had some additional commits with changes that were supposed to help with determining if it's supported by server but I've removed most of that because that had its own problems.

Reply:
OK, let's sort it out:
1. USING TIMEOUT is supported in every reasonable and supported Scylla version. Period. Let's not look into older versions.
2. With Cassandra, I still think catching the exception (which you should get right away) and retrying without USING TIMEOUT would be easiest.
2b. You should be able to distinguish between Scylla and Cassandra easily, and I assume the driver already has this distinction somewhere (no. of shards for example?) - and then use USING TIMEOUT only for Scylla (again, regardless of the version).

Reply:
This is what we already agreed on.
I don't understand what's the confusion here.